File: 프로그래머스/Level1/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    vector<int> tmp;
    
    for(int i=0; i<numbers.size()-1; i++){
        for(int j=i+1; j<numbers.size(); j++){
            tmp.push_back(numbers[i] + numbers[j]);
        }
    }
    
    sort(tmp.begin(), tmp.end());
    
    for(int i=0; i<tmp.size(); i++){
        answer.push_back(tmp[i]);
        while(tmp[i] == tmp[i+1]){ 
            i++;
        }
    }
    
    return answer;
}

File: 프로그래머스/Level1/두개 뽑아서 더하기.js
function solution(numbers) {
    let answer = [];
    let tmp = [];
    
    for(let i=0; i<numbers.length-1; i++){
        for(let j=i+1; j<numbers.length; j++){
            tmp.push(numbers[i] + numbers[j]);
        }
    }
    
    tmp.sort((a,b)=>a-b);
    
    for(let i=0; i<tmp.length; i++){
        answer.push(tmp[i]);
        
        while(tmp[i] == tmp[i+1]){
            i++;
        }
    }

    return answer;
}
